content/blog/2024/12/how-to-migrate-from-idf-bootloader-to-mcuboot-using-esp-self-reflasher/index.md

```diff
@@ -0,0 +1,271 @@
+    B -->|Reflashing process| C(MCUboot bootloader + NuttX application)
+```
+
+## ESP Self Reflasher component
+
+The [ESP Self-Reflasher](https://components.espressif.com/components/espressif/esp-self-reflasher/versions/1.0.0) is an ESP-IDF component that allows the device to be "reflashed" with new firmware. This is achieved through one of the following ways of operation: downloading the reflashing image or embedding the reflashing image. The component has examples for each that will be used on this guide.
+
+## Environment pre requisites
+
+Besides ESP-IDF ([Get started guide here](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/get-started/index.html)), the following are required:
+
+- ESP Self-Reflasher: clone it to `<IDF_DIR>/components` with Git:
+
+```bash
+cd <IDF_DIR>/components
+git clone https://github.com/espressif/esp-self-reflasher.git
+```
+
+- NuttX: in order to build the application that will be migrated to, set the NuttX workspace. [Get started guide here](https://developer.espressif.com/blog/nuttx-getting-started/).
+
+- MCUboot: in the case of building the bootloader standalone, follow the instructions [here](https://docs.mcuboot.com/readme-espressif.html).
+
+## Step-by-step Guide
+
+### Building the reflashing images (MCUboot and NuttX application)
+
+With the NuttX workspace ready, prepare the NSH (NuttShell) configuration with MCUboot compatibility, which is available under the `mcuboot_nsh` defconfig. This is the build that will be used as the *final target* of the reflashing process.
+
+```bash
+cd <NUTTX_DIR>
+./tools/configure.sh esp32-devkitc:mcuboot_nsh
+```
+
+---
+**Note**
+
+It is also possible to manually configure other NuttX applications for building it with MCUboot compatibility. Enter the menuconfig:
+
+```bash
+make menuconfig
+```
+
+Enter the menu `System Type -> Bootloader and Image Configuration` and enable the `Enable MCUboot-bootable format` option.
+
+---
+
+Ensure that the organization of image slots in flash is properly configured. It can be checked on the menu `System Type -> SPI Flash Configuration` from menuconfig. For this example using ESP32, the following is considered (note that the Bootloader offset may differ among other Espressif chip families):
+
+| REGION                           | OFFSET    | SIZE     |
+| -------------------------------- | --------- | -------- |
+| Bootloader                       | 0x001000  | 0x00F000 |
+| Application primary slot         | 0x010000  | 0x100000 |
+| Application secondary slot       | 0x110000  | 0x100000 |
+| Scratch                          | 0x210000  | 0x040000 |
+
+
+---
+**Note**
+
+If MCUboot is being build separately from NuttX build, be sure that the configuration for flash organization matches on both.
+
+---
+
+Save and exit the menuconfig.
+
+Build the MCUboot bootloader:
+
+```bash
+make -j bootloader
+```
+
+Now build the NuttX application:
+
+```bash
+make ESPTOOL_BINDIR=./ -s -j
+```
```

Comment:
This part looks like steps of the procedure. Please format these and the preceding steps so that they look like a procedure (not sure what the step numbers should be here, I randomly chose `4` to illustrate the formatting common for procedures):

````suggestion
4. Build the MCUboot bootloader:
  ```bash
  make -j bootloader
  ```
5. Now build the NuttX application:
  ```bash
  make ESPTOOL_BINDIR=./ -s -j
  ```
````